python/cudf/cudf/tests/test_groupby.py

```diff
@@ -789,24 +799,35 @@ def test_groupby_index_type():
 
 
 @pytest.mark.parametrize(
-    "interpolation", ["linear", "lower", "higher", "midpoint", "nearest"]
+    "interpolation", ["linear", "lower", "higher", "nearest", "midpoint"]
 )
-def test_groupby_quantile(interpolation):
+@pytest.mark.parametrize("q", [0.25, 0.4, 0.5, 0.7, 1])
+def test_groupby_quantile(interpolation, q):
     raw_data = {
-        "x": [1, 2, 3, 1, 2, 2, 1, None, 3, 2],
         "y": [None, 1, 2, 3, 4, None, 6, 7, 8, 9],
+        "x": [1, 2, 3, 1, 2, 2, 1, None, 3, 2],
     }
-    pdf = pd.DataFrame(raw_data)
+    # Pandas>0.25 now casts NaN in quantile operations as a float64
+    # # so we are filling with zeros.
+    pdf = pd.DataFrame(raw_data).fillna(0)
     gdf = DataFrame.from_pandas(pdf)
+
     pdg = pdf.groupby("x")
     gdg = gdf.groupby("x")
-    pdresult = pdg.quantile(interpolation=interpolation)
-    gdresult = gdg.quantile(interpolation=interpolation)
+
+    pdresult = pdg.quantile(q, interpolation=interpolation)
+    gdresult = gdg.quantile(q, interpolation=interpolation)
 
     # There's a lot left to add to python bindings like index name
     # so this is a temporary workaround
     pdresult = pdresult["y"].reset_index(drop=True)
     gdresult = gdresult["y"].reset_index(drop=True)
+
+    if q == 0.5 and interpolation == "nearest":
+        pytest.xfail(
+            "Pandas NaN Rounding will fail nearest interpolation at 0.5"
+        )
```

Comment:
Could you explain this expected failure?

Reply:
Yes sir. This is related to pandas new `groupby_quantile` implementation. 

https://github.com/pandas-dev/pandas/blob/0.25.x/pandas/_libs/groupby.pyx#L765

I am no expert on quantile but it appears to me that because they cast `NaN` as a `<float64_t>` now and potentially with how they are truncating the numbers, we just don't line up with their outputs with out current quantile implementation in some spots. I think this is a separate ticket and I know there has been other work going on on our quantile implementation so I wanted to sync with someone in the know now that we are back from holiday to see if this is already covered by someone else's work or if I need to write up an issue for it.

Reply:
Thanks for the explanation, makes sense to me.